pkgdb/src/registry/floxpkgs.cc

```diff
@@ -0,0 +1,696 @@
+   * attributes. */
+  flox::debugLog( "using our fetcher" );
+  auto                 ghAttrsAndOurAttrs = toGitHubAttrs( input.attrs );
+  nix::fetchers::Input asGithub           = input;
+  asGithub.attrs                          = ghAttrsAndOurAttrs.first;
+  auto ourAttrs                           = ghAttrsAndOurAttrs.second;
+
+  /* Wrap the GitHub flake ref in our rules processor. */
+  nix::FlakeRef nixpkgsRef = nix::FlakeRef::fromAttrs( asGithub.attrs );
+  auto          flakeDir   = flox::createWrappedFlakeDir( nixpkgsRef );
+  flox::debugLog( "created wrapped flake: path=" + std::string( flakeDir ) );
+
+  /* Compute the narHash of the wrapped flake. */
+  nix::StringSink sink;
+  nix::dumpPath( flakeDir, sink );
+  auto narHash = hashString( nix::htSHA256, sink.s );
+
+  /* Put the wrapped flake in the store. */
+  nix::ValidPathInfo info {
+    *store,
+    "source",
+    nix::FixedOutputInfo {
+      .method     = nix::FileIngestionMethod::Recursive,
+      .hash       = narHash,
+      .references = {},
+    },
+    narHash,
+  };
+  info.narSize = sink.s.size();
+  nix::StringSource source( sink.s );
+  store->addToStore( info, source );
+  nix::StorePath path = info.path;
+  flox::debugLog( "added filled out template flake to store: store_path="
+                  + std::string( path.to_string() ) );
+
+  /* Compute hashes for the rules file and processor so they can be included in
+   * the lockfile. */
+  auto floxNixpkgsInput   = asGithub;
+  floxNixpkgsInput.attrs  = fromGitHubAttrs( asGithub.attrs, ourAttrs );
+  floxNixpkgsInput.scheme = std::make_shared<FloxFlakeScheme>();
+  return std::pair<nix::StorePath, nix::fetchers::Input>( path,
+                                                          floxNixpkgsInput );
+}
+
+bool
+FloxFlakeScheme::hasAllInfo( const nix::fetchers::Input & ) const
+{
+  return true;
+}
+
+nix::ParsedURL
+FloxFlakeScheme::toURL( const nix::fetchers::Input & _input ) const
+{
+  auto              input = _input;
+  GitHubInputScheme githubScheme;
+  auto              ghAttrsAndOurAttrs = toGitHubAttrs( input.attrs );
+  input.attrs                          = ghAttrsAndOurAttrs.first;
+  auto url                             = githubScheme.toURL( input );
+  url.scheme                           = this->type();
+  return url;
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+std::string
+getRulesHash()
+{
+  auto rulesPath = getRulesFile();
+  auto rulesHash = nix::hashFile( nix::htSHA256, rulesPath );
+  return rulesHash.to_string( nix::Base16, false );
+}
+
+
+/* -------------------------------------------------------------------------- */
+
+std::string
+getRulesProcessorHash()
+{
+  static const std::string flakeTemplate =
+#include "./floxpkgs/flake.nix.in.hh"
+    ;
+  auto processorHash = nix::hashString( nix::htSHA256, flakeTemplate );
+  return processorHash.to_string( nix::Base16, false );
```

Comment:
suggestion: maybe also needs to be SRI?

Reply:
As far as I can tell you only need SRI when you're computing a nar hash? This hash is literally just stored in an attribute that no other code besides ours uses so I'm not sure how much it really matters.